### PR TITLE
MSA Integration Tests Part 1

### LINKED ISSUE #695: Integration Tests: Token Holder MSA
Happy path for someone who uses tokens.

## Extrinsics to touch
- [ ]  create()
- [ ]  addPublicKeyToMsa(msaOwnerPublicKey, msaOwnerProof, newKeyOwnerProof, addKeyPayload)
- [ ]  deleteMsaPublicKey(publicKeyToDelete)
- [ ]  revokeSchemaPermissions(providerMsaId, schemaIds)
- [ ]  revokeDelegationByDelegator(providerMsaId)
- [ ]  grantDelegation(delegatorKey, proof, addProviderPayload)
- [ ]  retireMsa()

### PR DESCRIPTION
**Pull request by dopatraman**
# Goal
The goal of this PR is to test an MSA workflow that involves:
1. Creating an MSA
2. Adding a public key
3. Deleting a public key
4. Creating a Schema
5. Retiring the MSA
6. Creating a provider
7. Granting a delegation
8. Revoking a delegation

Closes #695

## Notes
This PR introduces a `scenarios/` dir that contains cross-pallet tests. If you want to change this structure, leave a note.

# Checklist
- [x] Tests added

### CODE REVIEW

File: integration-tests/scaffolding/extrinsicHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { firstValueFrom, filter } from "rxjs";
+import { EventMap, groupEventsByKey, Sr25519Signature } from "./helpers";
+
+type AddKeyData = {
+    msaId?: any;
+    expiration?: any;
+    newPublicKey?: any;
+}
+
+/** Schema Extrinsics */
```
Comment (dopatraman) [inline integration-tests/scaffolding/extrinsicHelpers.ts:13]: If we continue to use helpers like this, we can separate these helpers by pallet.

File: integration-tests/scaffolding/apiConnection.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import { KeyringPair } from "@polkadot/keyring/types";
+import { firstValueFrom } from "rxjs";
+
+export async function connect(providerUrl): Promise<ApiRx> {
+    const provider = new WsProvider(providerUrl);
+    const apiObservable = ApiRx.create({ provider, ...options });
+    return firstValueFrom(apiObservable);
+}
+
+export function createKeys(uri: string): KeyringPair {
```
Comment (shannonwells) [inline integration-tests/scaffolding/apiConnection.ts:12]: ❤️

File: integration-tests/scaffolding/extrinsicHelpers.ts (hunk 1 above)
Comment (shannonwells) [reply integration-tests/scaffolding/extrinsicHelpers.ts:15]: These look great!

File: scripts/kill_freq.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/zsh
+
+set -e
+
+PID="$(ps aux | grep target/release/frequency | grep -v grep | xargs | awk '{print $2}')"
+
+if ! [ -z $PID ]
+then
+    kill -9 $PID
+    echo "Frequency is dead."
```
Comment (shannonwells) [inline scripts/kill_freq.sh:10]: this script sorely needs a 💀 emoji

**Review by shannonwells — APPROVED** [review]
This is really awesome. Nice work!

**Review by aramikm — APPROVED** [review]
Looks good! Added one comment.

File: integration-tests/msa/createMsa.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    })
+
+    after(() => {
+        api.disconnect()
+    })
+
+    // NOTE: We will need a sustainable way to create new keys for every test,
+    // since there is only one node instance per test suite.
+    it("should successfully create an MSA account", async () => {
+        keys = createKeys("//Alice")
```
[the remark this answers is not in the document]
Comment (dopatraman) [reply integration-tests/msa/createMsa.test.ts:27]: great idea. let me investigate.

**Pull request merged**